XRTK-Core/Packages/com.xrtk.core/Providers/Controllers/Hands/BaseHandController.cs

```diff
@@ -107,6 +99,8 @@ private void UpdateJoints(HandData handData)
 
         private void UpdateBounds()
         {
+            var handControllerDataProvider = (IMixedRealityHandControllerDataProvider)ControllerDataProvider;
```

Comment:
We might want to locally cache this cast instead of doing it every update.

Reply:
Should be okay, it's just a simple type cast. Nothing  too crazy